src/Umbraco.Core/Composing/TypeFinder.cs

```diff
@@ -49,13 +49,25 @@ public class TypeFinder : ITypeFinder
     private string[]? _assembliesAcceptingLoadExceptions;
     private volatile HashSet<Assembly>? _localFilteredAssemblyCache;
 
+    [Obsolete("Please use constructor that takes additionalExlusionAssemblies, scheduled for removal in v14")]
     public TypeFinder(ILogger<TypeFinder> logger, IAssemblyProvider assemblyProvider, ITypeFinderConfig? typeFinderConfig = null)
     {
         _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         _assemblyProvider = assemblyProvider;
         _typeFinderConfig = typeFinderConfig;
     }
 
+    public TypeFinder(ILogger<TypeFinder> logger, IAssemblyProvider assemblyProvider, string[]? additionalExlusionAssemblies, ITypeFinderConfig? typeFinderConfig = null)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _assemblyProvider = assemblyProvider;
+        _typeFinderConfig = typeFinderConfig;
+        if (additionalExlusionAssemblies is not null)
+        {
+            KnownAssemblyExclusionFilter = KnownAssemblyExclusionFilter.Concat(additionalExlusionAssemblies).ToArray();
```

Comment:
What happens with duplicates ? Should we call a unique on it?

Reply:
Skipping duplicates when combining two sequences can be done using [`Union`](https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.union?view=net-7.0):
```suggestion
            KnownAssemblyExclusionFilter = KnownAssemblyExclusionFilter.Union(additionalExlusionAssemblies).ToArray();
```

This even has an overload that takes an `IEqualityComparer<>`, so you can pass `StringComparer.OrdinalIgnoreCase` into it to make it ignore casing differences.